Index: gecode/support/timer.hh
===================================================================
--- gecode/support/timer.hh	(revision 9681)
+++ gecode/support/timer.hh	(working copy)
@@ -80,7 +80,15 @@
       timeval t1, t;
       if (gettimeofday(&t1, NULL))
         throw OperatingSystemError("Timer::stop[gettimeofday]");
-      timersub(&t1, &t0, &t);
+
+      // timersub(&t1, &t0, &t);
+      t.tv_sec = t1.tv_sec - t0.tv_sec;
+      t.tv_usec = t1.tv_usec - t0.tv_usec;
+      if (t.tv_usec < 0) {
+        t.tv_sec--;
+        t.tv_usec += 1000000;
+      }
+      
       return (static_cast<double>(t.tv_sec) * 1000.0) + 
         (static_cast<double>(t.tv_usec)/1000.0);
 #elif defined(GECODE_USE_CLOCK)